uaclient/contract.py

```diff
@@ -368,80 +369,87 @@ def _update_contract_machine(
+        }
+        backcompat_data = _support_old_machine_info(data)
         url = API_V1_UPDATE_CONTRACT_MACHINE.format(
-            contract=contract_id, machine=data["machineId"]
+            contract=contract_id, machine=machine_id
         )
         response = self.request_url(
-            url, headers=headers, method="POST", data=data
+            url, headers=headers, method="POST", data=backcompat_data
         )
         if response.code != 200:
             raise exceptions.ContractAPIError(
                 url, response.code, response.body
             )
         if response.headers.get("expires"):
             response.json_dict["expires"] = response.headers["expires"]
-        machine_id = response.json_dict.get("machineTokenInfo", {}).get(
-            "machineId", data.get("machineId")
-        )
         return response.json_dict
 
-    def update_files_after_machine_token_update(
-        self, response: Dict[str, Any]
-    ):
-        self.cfg.machine_token_file.write(response)
-        system.get_machine_id.cache_clear()
-        data = self._get_platform_data(None)
-        machine_id = response.get("machineTokenInfo", {}).get(
-            "machineId", data.get("machineId")
-        )
-        self.cfg.write_cache("machine-id", machine_id)
-
-    def _get_platform_data(self, machine_id):
-        """Return a dict of platform-related data for contract requests"""
-        if not machine_id:
-            machine_id = system.get_machine_id(self.cfg)
-        return {
-            "machineId": machine_id,
+    def _get_activity_info(self):
+        """Return a dict of activity info data for contract requests"""
+        machine_info = {
+            "distribution": system.get_release_info().distribution,
+            "kernel": system.get_kernel_info().uname_release,
+            "series": system.get_release_info().series,
             "architecture": system.get_dpkg_arch(),
-            "os": {
-                "type": "Linux",
-                "distribution": system.get_release_info().distribution,
-                "release": system.get_release_info().release,
-                "series": system.get_release_info().series,
-                "version": system.get_release_info().pretty_version,
-                "kernel": system.get_kernel_info().uname_release,
-                "virt": system.get_virt_type(),
-            },
+            "desktop": system.is_desktop(),
+            "virt": system.get_virt_type(),
+            "clientVersion": version.get_version(),
         }
 
-    def _get_platform_basic_info(self):
-        """Return a dict of platform basic info for some contract requests"""
+        if _is_attached(self.cfg).is_attached:
+            enabled_services = _enabled_services(self.cfg).enabled_services
+            attachment_data = attachment_data_file.read()
+            activity_info = {
+                "activityID": self.cfg.machine_token_file.activity_id,
```

Comment:
I think this needs a 
```suggestion
                "activityID": self.cfg.machine_token_file.activity_id or system.get_machine_id(cfg)
```